mailer.rst

```diff
@@ -1459,6 +1459,44 @@ and their priorities:
 
     $ php bin/console debug:event-dispatcher "Symfony\Component\Mailer\Event\MessageEvent"
 
+QueuingMessageEvent
+~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\Mailer\\Event\\QueuingMessageEvent`
+
+``QueuingMessageEvent`` allows to add some logic before the email is sent to
+the Messenger bus (this event is not dispatched when no bus is configured); it
+extends ``MessageEvent`` to allow adding Messenger stamps to the Messenger
+message sent to the bus::
+
+    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+    use Symfony\Component\Mailer\Event\QueuingMessageEvent;
+    use Symfony\Component\Mime\Email;
+
+    public function onMessage(QueuingMessageEvent $event): void
+    {
+        $message = $event->getMessage();
+        if (!$message instanceof Email) {
+            return;
+        }
+        // do something with the message
+
+        // and/or add some Messenger stamps
+        $event->addStamp(new SomeMessengerStamp());
+    }
+
+This event lets listeners do something before a message is sent to the queue
+(like adding stamps or logging) but any changes to the message or the envelope
+are discarded. To change the message or the envelope, listen to
+``MessageEvent`` instead.
```

Comment:
This paragraph seems to contradict the "do something with the message" comment in the code example above.

Is it saying "you can change the messenger message, but not the email" or is this a copy-paste error?

Reply:
You can change the message if you want to, it will just be ignored. That's useful when you want to get errors early on (before sending things to the queue) like rendering the message to be sure there are no errors in templates. I have added a parenthesis with logging t make things a bit clearer about the primary use cases.